spec/models/book_spec.rb

```diff
@@ -39,9 +39,9 @@
 
   describe "#books_borrowed" do
     let(:book) {FactoryBot.create :book, category: FactoryBot.create(:category)}
-    let(:request) {FactoryBot.create :request, user: FactoryBot.create(:user), status: Request.statuses[:approved]}
+    let!(:request) {FactoryBot.create :request, user: FactoryBot.create(:user),
```

Comment:
`let` và 'let!' khác nhau chỗ nào em. Anh thấy trước em dùng 'let' giờ đổi lại 'let!'

Reply:
let là lazy evaluation nghĩa chỉ đc khai báo vào lần đầu tiên đc gọi trong một example
let! thì như 1 before each đc gọi trước mỗi example
 ở đây em đổi thành let! vì code trước e tạo request_book từ request trong khối it
mà giờ khi tạo request phải có request_book...nên em đổi thành let! vì trong khối it ko có gọi nữa